nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,467 @@
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+
+            # bagel emoji if at least one participant has been processed
+            # through all pipelines
+            if (row[bagel_cols] > 0).all():
+                reward_emojis.append(bagel_emoji)
+
+            # confetti_emoji if processing is complete for all
+            # participants and pipelines
+            all_equal = False
+            if (row[status_count_cols] > 0).all():
+                row_vals = row[status_count_cols].values
+                all_equal = all([x == row_vals[0] for x in row_vals])
+                if all_equal:
+                    reward_emojis.append(confetti_emoji)
+
+            # Cannot nest this in prior if statement because
+            # some pipeline count might be zero
+            # party_popper_emoji if at least one pipeline is complete
+            if (not all_equal) & (row[bagel_cols] == row[manifest_col]).any():
+                reward_emojis.append(party_popper_emoji)
+            # rocket_emoji if at least one pipeline is in progress
+            elif (not all_equal) & (row[bagel_cols] > 0).any():
+                reward_emojis.append(rocket_emoji)
+
+            # Additional rewards based on quantity
+            # if (row[bagel_cols] > high_throughput).any():
+            #     reward_emojis.append(mate_emoji)
+            # elif (row[bagel_cols] > medium_throughput).any():
+            #     reward_emojis.append(shaved_ice_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _df_to_table(self, status_df: pd.DataFrame, legend_dict: dict):
+        """Convert a pandas.DataFrame obj into a rich.Table obj."""
+        df = status_df.copy().reset_index()
+        df = df.sort_values(by=self.manifest.col_session_id)
+
+        console = Console()
+
+        crossed_fingers_emoji = "🤞"
+        column_colors = [
+            "cyan",
+            "magenta",
+            "yellow",
+            "green",
+            "deep_sky_blue3",
+            "deep_pink2",
+            "rosy_brown",
+        ]
+
+        # Initiate a Table instance
+        title = (
+            "Participant counts by session at each Nipoppy checkpoint"
+            + f" {crossed_fingers_emoji} {crossed_fingers_emoji}"
+        )
+
+        table = Table(title=title, collapse_padding=False)
+
+        # check if number of colors match the number of columns
+        n_colors = len(column_colors)
+        n_columns = len(df.columns)
+        if n_colors < n_columns:  # cycle through the colors
+            column_colors = (
+                column_colors * (n_columns // n_colors + 1)
+                + column_colors[: n_columns % n_colors]
+            )
+
+        for column, col_color in zip(df.columns, column_colors):
+            table.add_column(
```

Comment:
I *think* this should work, and it would simplify the code

```suggestion
        n_colors = len(column_colors)
        for i_col, column in enumerate(df.columns):
            col_color = column_colors[i_col % n_colors]
            table.add_column(
```